pkg/controller/certificates/util.go

```diff
@@ -178,6 +182,29 @@ func RequestMatchesSpec(req *cmapi.CertificateRequest, spec cmapi.CertificateSpe
 	return violations, nil
 }
 
+// ExtraNameFieldEqual will check the Subject to see that it has the expected extra Subject fields, if any are defined.
+// Note the "ExtraNames" of the CertificateSpec end up in the "Names" (Subject) of the CertificateRequest.
+func ExtraNameFieldEqual(requestNames []pkix.AttributeTypeAndValue, subjectExtraNames []cmapi.ExtraName) bool {
+	// no need to look further if no extra names were requested
+	if len(subjectExtraNames) == 0 {
+		return true
+	}
+
+	for _, requestAttributeTypeAndValue := range requestNames {
```

Comment:
Just wanted to add a comment here to have someone take a second look at this part as it's probably the most critical piece of logic.

Reply:
Please write some unit tests for this function - I feel like there's a number of cases where this may not return as expected (it seems like you're returning as soon as the first matching extraName is found, but that doesn't account for any of the _other_ extraNames if there are multiple).

Also, should the _order_ of these matter?

Reply:
That is true - it's probably more correct to do:

```
if subjectExtraName.Value != requestAttributeTypeAndValue.Value {
return false
}
```

I tried adding some unit tests in a new file, but the bazel build failed, so I deleted them. I did add them to some sort of bazel manifest where it seems to list all of the files in that folder/module because it looked like that's what one was supposed to do, but it still failed. I'll add back the unit tests along with more of them and ask for help in the slack channel if it keeps failing (unless you know off-hand why that might be).

Reply:
Did you run `./hack/update-bazel.sh` after adding the new file? 😄

Reply:
I did, I actually ran both update-bazel.sh and update-all.sh. I tried update-all after update-bazel didn't do the trick. I'll try again when I make these updates. There was some weird build stuff going on at the time.